From fa87ade8b285a75c03c9cd064181226dbb0ba1dc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B8rgen=20S=2E=20Dokken?= <dokken@simula.no>
Date: Thu, 25 Sep 2025 12:59:24 +0000
Subject: [PATCH] Speed up real space tests

---
 tests/test_real_functionspace.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_real_functionspace.py b/tests/test_real_functionspace.py
index 02258b1..4bb7a02 100644
--- a/tests/test_real_functionspace.py
+++ b/tests/test_real_functionspace.py
@@ -89,7 +89,7 @@ def test_real_function_space_vector(cell_type, dtype):
 @pytest.mark.parametrize("tensor", [0, 1, 2])
 @pytest.mark.parametrize("degree", range(1, 5))
 def test_singular_poisson(tensor, degree, dtype):
-    M = 25
+    M = 9
 
     rtype, stype = dtype
     mesh = dolfinx.mesh.create_unit_square(
-- 
2.43.0

